src/QMCDrivers/WFOpt/QMCCostFunctionBase.cpp

```diff
@@ -605,69 +605,59 @@ void QMCCostFunctionBase::updateXmlNodes()
     addCJParams(acontext, "//jastrow");
     xmlXPathFreeContext(acontext);
   }
-  //     Psi.reportStatus(app_log());
-  std::map<std::string, xmlNodePtr>::iterator pit(paramNodes.begin()), pit_end(paramNodes.end());
-  while (pit != pit_end)
+  for (auto& [pname, pptr] : paramNodes)
```

Comment:
Cool. we are taking advantage of C++17. Does adding const apply here?

Reply:
The map key is already const, though we can also add const to the value as well without changing code.